src/utils/geometry.rs

```diff
@@ -725,6 +725,35 @@ impl<N: Coordinate, Kind> Rectangle<N, Kind> {
     }
 }
 
+impl<Kind> Rectangle<f64, Kind> {
+    /// Convert to i32 for integer-space manipulations by rounding float values
+    #[inline]
+    pub fn to_i32_round<N: Coordinate>(self) -> Rectangle<N, Kind> {
+        Rectangle {
+            loc: self.loc.to_i32_round(),
+            size: self.size.to_i32_round(),
+        }
+    }
+
+    /// Convert to i32 for integer-space manipulations by flooring float values
+    #[inline]
+    pub fn to_i32_floor<N: Coordinate>(self) -> Rectangle<N, Kind> {
+        Rectangle {
+            loc: self.loc.to_i32_floor(),
+            size: self.size.to_i32_floor(),
+        }
+    }
+
+    /// Convert to i32 for integer-space manipulations by ceiling float values
+    #[inline]
+    pub fn to_i32_ceil<N: Coordinate>(self) -> Rectangle<N, Kind> {
+        Rectangle {
+            loc: self.loc.to_i32_ceil(),
+            size: self.size.to_i32_ceil(),
+        }
+    }
+}
+
```

Comment:
That is a good catch actually, because I think I am already mindlessly using them in their "spirit" instead of what they are actually doing.

But wouldn't it just suffice for `to_i32_floor`  to change the location to `self.loc.to_i32_ceil()` and vise-versa?

Reply:
A counter-example (looking only at one coordinate): take a rectangle spanning from 1.9 to 5.1 that we want to "ceil in spirit". 

Its location is 1.9, and its size is 3.2. Just doing floor/ceil would produce an integer rectangle of location 1 and size 4, meaning a rectangle that goes from 1.0 to 5.0, while proper "ceil in spirit" should produce a rectangle going from 1.0 to 6.0.

Reply:
Ah right, the size needs to account for the shift of the location. Makes sense

Reply:
Should be fixed in #434. If not the discussion should be moved there.